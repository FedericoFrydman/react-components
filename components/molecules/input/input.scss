@import '../../assets/styles/fonts.scss';
@import '../../assets/styles/colors.scss';

@mixin createBorder($color) {
  border: 1px solid $color;
  -webkit-border-color: none;
  border-radius: var(--brand-border-radius);
  -moz-box-sizing: border-box; /* Firefox */
  -webkit-box-sizing: border-box; /* Safari/Chrome */
  box-sizing: border-box;
}

@mixin focusState($color) {
  border-color: $color;
}

.lsux-input-container {
  flex: 1;
  position: relative;

  &__input {
    background-color: $N00;
    -moz-box-sizing: border-box; /* Firefox */
    -webkit-box-sizing: border-box; /* Safari/Chrome */
    box-sizing: border-box;
    color: $N800;
    @include createBorder($N350);
    font-style: normal;
    height: 44px;
    letter-spacing: -0.011em;
    line-height: 22px;
    outline: none;
    padding-left: 16px;
    padding-right: 16px;
    -moz-transition: border-color ease-in-out 0.15s; /* Firefox */
    -webkit-transition: border-color ease-in-out 0.15s; /* Safari/Chrome */
    transition: border-color ease-in-out 0.15s;
    width: 100%;

    &:focus:not(:read-only) {
      @include focusState(var(--brand-color-prim200));
      outline: none;
    }

    &--warning {
      background: $Y10;
      outline: none;
      @include createBorder($Y200);

      &:focus {
        @include focusState($Y200);
      }
    }

    &--valid {
      @include createBorder($G200);
      outline: none;

      &:focus {
        @include focusState($G200);
      }
    }

    &--invalid {
      background: $R10;
      outline: none;
      @include createBorder($R200);

      &:focus {
        @include focusState($R200);
      }

      &:not(:focus) {
        color: $R200;
        & + .lsux-text {
          color: $R200;
        }
      }
    }

    &--icon-left {
      padding-left: 40px;
    }

    &--icon-right {
      padding-right: 40px;
    }

    &:disabled {
      background-color: $N50;
      color: $N350;
      cursor: not-allowed;
    }

    &:-moz-read-only {
      border: none;
      color: $N700;
    }

    &:read-only {
      border: none;
      color: $N700;
    }

    &::-webkit-inner-spin-button {
      background: $N100;
      border: 1px solid $N500;
      border-radius: var(--brand-border-radius);
      height: 40px;
      margin: 0;
      opacity: 1;
    }

    &::-webkit-calendar-picker-indicator {
      background: none;
      cursor: pointer;
      display: block;
      height: 20px;
      margin-right: 10px;
      outline: none;
      width: 20px;
    }

    &::placeholder {
      align-items: center;
      color: $N500;
      display: flex;
      right: 8px;
      top: 8px;
    }

    &--date-number::-webkit-inner-spin-button,
    &--date-number::-webkit-outer-spin-button {
      -webkit-appearance: none;
      border: none;
      margin: 0;
    }
  }

  .lsux-input__icon {
    top: 8px;
    &-left {
      left: 8px;
    }
    &-right {
      right: 8px;
    }
    &-hide-icon {
      border-radius: var(--brand-border-radius);
      bottom: 0;
      position: absolute;
      right: 0;
      top: 0;
      width: 34px;
      img {
        top: 12px;
      }
    }
    &-search-icon {
      left: 8px;
      position: absolute;
      top: 8px;
    }
  }

  .lsux-button {
    background-color: transparent;
  }

  .lsux-icon {
    cursor: pointer;
    position: absolute;
    right: 8px;
    top: 10px;
  }
}
