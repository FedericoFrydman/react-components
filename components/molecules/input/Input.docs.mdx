import { ArgsTable, Story, Canvas, Stories } from '@storybook/addon-docs';
import { Alert } from '../alert/Alert';
import { Input } from './Input';

# Input Props

**This documentantion is just to display the input props where you can interact with them and see how it behaves. For implementation please use [FormField](/?path=/docs/forms-form-fields--input-field) docs to gide.**

<Alert title="Dev Note" severity="warning" icon="alert_warning">
  The Input component must be used as a child of a FormField component.
</Alert>

<p style={{ margin: 0 }}>&nbsp;</p>

<ArgsTable of={Input} />

**About the input:**

- Input allows people to enter data for further use in your product, or to configure options. Inputs and forms are used to get information and guide people with minimal fuss.
- Input should always be accompanied by a label for accessibility purposes with the exception of Search (for this reason, the default text color for Search is N1000 for color contrast.
- Input has a HTML property call autoComplete it should be equal to "off" in order to not be auto completed.

## Example Code

```tsx
<FormField id="name-input" status={name.status} validationHint={name.hint} label="First Name" type="input" required>
  <Input
    name="name-input"
    onChange={handleInputChange}
    placeholder="First Name"
    value={name.value}
    type="text"
    required
  />
</FormField>
```

<Stories />
