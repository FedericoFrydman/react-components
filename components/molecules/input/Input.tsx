import { cloneElement, DetailedHTMLProps, forwardRef, ReactElement, Ref, useEffect, useState } from 'react';
import NumberFormat, { NumberFormatProps } from 'react-number-format';
import cn from 'classnames';

import { Button } from '../button/Button';
import { Icon, IconNames, IconProps } from '../../atoms/icon/Icon';
import { IconColors } from '../../../components/assets/icons';
import './input.scss';

export type InputValidationStatus = '' | 'valid' | 'warning' | 'invalid';

export type InputType =
  | 'color'
  | 'date'
  | 'datetime-local'
  | 'email'
  | 'hidden'
  | 'month'
  | 'number'
  | 'password'
  | 'range'
  | 'search'
  | 'tel'
  | 'text'
  | 'time'
  | 'url'
  | 'week'
  | 'currency';

export interface InputProps extends Omit<React.HTMLProps<HTMLInputElement>, 'type'> {
  /**
   * Triggers error | warning | success styles on the component. Important for accessibility.
   */
  status?: InputValidationStatus;
  /**
   * Determines the type of the Input.
   * @default text
   */
  type?: InputType;
  /**
   * Determines which icon should be shown on the right; if a password type this property is managed internally with the hide/show icons.
   * @deprecated Please use `iconLeft` and `iconRight` instead.
   */
  icon?: IconNames;
  /**
   * Icon on the left side of the input
   */
  iconLeft?: ReactElement<IconProps>;
  /**
   * Icon on the right side of the input
   */
  iconRight?: ReactElement<IconProps>;
  /**
   * Applies a color to the icon.  This could be for the icon on the right or a search icon when type is search.
   * @default BRAND200
   */
  iconColor?: IconColors;
  /**
   * Custom class names for the Input.
   */
  classNames?: cn.Argument;
  /**
   * Format uses the NumberFormat component instead of Input. It allows for input masking.
   */
  format?: string;
  /**
   * Placeholder value for the input field.
   */
  placeholder?: string;
  /**
   * Prefix value for the input field. Only displays when `NumberFormat` component is rendered.
   */
  prefix?: string;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLInputElement>;
}

const InputComponent = (
  {
    status = '',
    type = 'text',
    icon,
    iconColor = 'BRAND200',
    format,
    placeholder,
    prefix,
    iconLeft,
    iconRight = icon ? <Icon name={icon} color={iconColor} /> : null,
    classNames = [],
    ...props
  }: InputProps,
  ref: InputProps['ref'],
) => {
  const [iconColorState, setIconColorState] = useState(iconColor);
  const [passwordVisible, setPasswordVisible] = useState(false);

  const classInputContainer = `lsux-input-container`;
  const cnInputContainer = cn(classInputContainer);

  const classInput = 'lsux-input-container__input';
  const cnInput = cn(
    {
      [`${classInput}`]: true,
      [`${classInput}--${status}`]: status,
      [`${classInput}--icon-left`]: type === 'search' || iconLeft,
      [`${classInput}--icon-right`]: type === 'password' || iconRight,
    },
    classNames,
  );

  const classInputIcon = 'lsux-input__icon';

  useEffect(() => {
    switch (status) {
      case 'invalid': {
        setIconColorState('R200');
        break;
      }
      case 'warning': {
        setIconColorState('Y200');
        break;
      }
      case 'valid': {
        setIconColorState('G200');
        break;
      }
      default: {
        setIconColorState(iconColor);
        break;
      }
    }
  }, [status, iconColor]);

  if (type === 'password' && !iconColor) {
    setIconColorState('BRAND200');
  }

  if (iconLeft) {
    iconLeft = cloneElement(iconLeft, {
      classNames: [iconLeft.props?.classNames, classInputIcon, `${classInputIcon}-left`],
      color: iconColorState,
    });
  }

  if (iconRight) {
    iconRight = cloneElement(iconRight, {
      classNames: [iconRight.props?.classNames, classInputIcon, `${classInputIcon}-right`],
      color: iconColorState,
    });
  }

  if (type === 'search') {
    iconLeft = (
      <Icon
        name="interface_search_magnifying_glass"
        classNames={[classInputIcon, `${classInputIcon}-search-icon`]}
        color={iconColorState}
      />
    );
  }

  if (type != 'password' && icon) {
    iconRight = <Icon name={icon} color={iconColorState} classNames={classInputIcon} />;
  }

  const handleIconClick = () => {
    setPasswordVisible(!passwordVisible);
  };

  const { ref: removedRef, ...other } = props;

  // Set default format value for `tel` input
  if (type === 'tel' && !format) {
    format = '(###) ###-####';
  }

  // Set default placeholder value for `tel` input
  if (type === 'tel' && !placeholder) {
    placeholder = '(___) ___-____';
  }

  const content = (
    <div className={cnInputContainer}>
      {iconLeft}
      {format && (type === 'password' || type === 'tel' || type === 'text') && (
        <NumberFormat
          className={cnInput}
          format={format}
          mask="_"
          placeholder={placeholder}
          prefix={prefix}
          type={type}
          {...(props as NumberFormatProps)}
        />
      )}
      {type === 'currency' && (
        <NumberFormat
          allowLeadingZeros={false}
          className={cnInput}
          format={format}
          placeholder={placeholder ?? `${prefix ?? '$'}0.00`}
          prefix={prefix ?? '$'}
          thousandSeparator={true}
          type="text"
          {...(props as NumberFormatProps)}
        />
      )}
      {!format && type !== 'currency' && (
        <input
          {...(props as DetailedHTMLProps<unknown, unknown>)}
          type={type === 'password' && passwordVisible ? 'text' : type}
          placeholder={placeholder}
          className={cnInput}
          ref={ref}
        />
      )}
      {type === 'password' && (
        <Button
          classNames={[classInputIcon, `${classInputIcon}-hide-icon`]}
          iconColor={iconColorState}
          iconLeft={passwordVisible ? 'action_eye_closed' : 'action_eye_open'}
          onClick={handleIconClick}
          variant="tertiary"
        />
      )}
      {type !== 'password' && iconRight}
    </div>
  );

  return content;
};

export const Input = forwardRef<HTMLInputElement, InputProps>(InputComponent);
