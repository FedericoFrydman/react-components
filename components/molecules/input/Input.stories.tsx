import { FormEvent, useState } from 'react';
import { Story } from '@storybook/react';

import { Input, InputProps } from './Input';
import docs from './Input.docs.mdx';
import { Container } from '../../../components/atoms/container/Container';
import { ICON_COLORS_V2, ICON_NAMES } from '../../../components/assets/icons';
import { Icon } from 'components/atoms/icon/Icon';

const InputStory: Story<InputProps> = (args: InputProps) => {
  const [, setValue] = useState(args.value);
  const handleOnChange = (e: FormEvent<HTMLInputElement>) => {
    setValue((e.target as HTMLInputElement).value);
  };

  return (
    <Container classNames={['m-4']}>
      <Input {...args} onChange={handleOnChange} />
    </Container>
  );
};

export const Default = InputStory.bind({});
Default.args = {
  autoCapitalize: 'words',
  defaultValue: 'Joe',
  disabled: false,
  id: 'base-input',
  name: 'base-input',
  placeholder: 'First Name',
  readOnly: false,
};

export const TypeColor = InputStory.bind({});
TypeColor.args = {
  disabled: false,
  id: 'input-color',
  name: 'input-color',
  type: 'color',
};
export const TypeDate = InputStory.bind({});
TypeDate.args = {
  disabled: false,
  id: 'input-date',
  name: 'input-date',
  type: 'date',
};
export const TypeDateTime = InputStory.bind({});
TypeDateTime.args = {
  disabled: false,
  id: 'input-datetime',
  name: 'input-datetime',
  type: 'datetime',
};
export const TypeEmail = InputStory.bind({});
TypeEmail.args = {
  disabled: false,
  id: 'input-email',
  name: 'input-email',
  type: 'email',
};
export const TypeMonth = InputStory.bind({});
TypeMonth.args = {
  disabled: false,
  id: 'input-month',
  name: 'input-month',
  type: 'month',
};
export const TypeNumber = InputStory.bind({});
TypeNumber.args = {
  disabled: false,
  id: 'input-number',
  name: 'input-number',
  type: 'number',
};
export const TelReadOnly = InputStory.bind({});
TelReadOnly.args = {
  id: 'phonenumber-input-readonly',
  placeholder: '1234567890',
  readOnly: true,
};
export const TypePassword = InputStory.bind({});
TypePassword.args = {
  disabled: false,
  id: 'input-password',
  name: 'input-password',
  type: 'password',
};
export const TypeRange = InputStory.bind({});
TypeRange.args = {
  disabled: false,
  id: 'input-range',
  name: 'input-range',
  type: 'range',
};
export const TypeSearch = InputStory.bind({});
TypeSearch.args = {
  disabled: false,
  id: 'input-search',
  name: 'input-search',
  type: 'search',
};
export const TypeTel = InputStory.bind({});
TypeTel.args = {
  disabled: false,
  id: 'input-tel',
  name: 'input-tel',
  type: 'tel',
};
export const TypeText = InputStory.bind({});
TypeText.args = {
  disabled: false,
  id: 'input-text',
  name: 'input-text',
  type: 'text',
};
export const TypeTime = InputStory.bind({});
TypeTime.args = {
  disabled: false,
  id: 'input-time',
  name: 'input-time',
  type: 'time',
};
export const TypeURL = InputStory.bind({});
TypeURL.args = {
  disabled: false,
  id: 'input-url',
  name: 'input-url',
  type: 'url',
};
export const TypeWeek = InputStory.bind({});
TypeWeek.args = {
  disabled: false,
  id: 'input-week',
  name: 'input-week',
  type: 'week',
};

export const TypeCurrency = InputStory.bind({});
TypeCurrency.args = {
  id: 'currency-input',
  type: 'currency',
};

export const Numeric = InputStory.bind({});
Numeric.args = {
  id: 'numeric-input',
  name: 'numeric-input',
  placeholder: '0',
  type: 'number',
};

export const ValidationStatusInvalid = InputStory.bind({});
ValidationStatusInvalid.args = {
  id: 'invalid-input',
  name: 'invalid-input',
  placeholder: 'Invalid input',
  required: true,
  status: 'invalid',
  value: 'Example Text',
};

export const ValidationStatusWarning = InputStory.bind({});
ValidationStatusWarning.args = {
  id: 'warning-input',
  name: 'warning-input',
  placeholder: 'Warning input',
  status: 'warning',
  value: 'Example Text',
};

export const ValidationStatusSuccess = InputStory.bind({});
ValidationStatusSuccess.args = {
  id: 'Success-input',
  name: 'Success-input',
  placeholder: 'Success input',
  status: 'valid',
  value: 'Example Text',
};

export const Disabled = InputStory.bind({});
Disabled.args = {
  disabled: true,
  id: 'disabled-input',
  name: 'disabled-input',
  placeholder: 'Disabled input',
};

export const ReadOnly = InputStory.bind({});
ReadOnly.args = {
  id: 'read-only-input',
  name: 'read-only-input',
  placeholder: 'Read only input',
  readOnly: true,
  value: 'Some read only value',
};

export const CustomIconLeft = InputStory.bind({});
CustomIconLeft.args = {
  iconLeft: <Icon name="interface_shopping_bag_02" />,
  id: 'read-only-input',
  name: 'read-only-input',
  placeholder: 'Some read only placeholder',
  readonly: false,
};

export const CustomIconRight = InputStory.bind({});
CustomIconRight.args = {
  iconRight: <Icon name="interface_shopping_bag_02" />,
  id: 'read-only-input',
  name: 'read-only-input',
  placeholder: 'Some read only placeholder',
  readonly: false,
};

export default {
  argTypes: {
    format: { control: { type: 'text' } },
    icon: { control: { type: 'select' }, options: ICON_NAMES },
    iconColor: {
      control: {
        type: 'select',
      },
      options: ICON_COLORS_V2,
    },
    placeholder: { control: { type: 'text' } },
    prefix: { control: { type: 'text' } },
    status: { control: { type: 'select' }, options: ['', 'valid', 'warning', 'invalid'] },
    type: {
      control: {
        options: [
          'color',
          'date',
          'datetime-local',
          'email',
          'hidden',
          'month',
          'number',
          'password',
          'range',
          'search',
          'tel',
          'text',
          'time',
          'url',
          'week',
          'currency',
        ],
        type: 'select',
      },
    },
    value: { control: { type: 'text' } },
  },
  component: InputStory,
  parameters: {
    docs: { page: docs },
  },
  title: 'Controls/Form Fields/Input Props',
};
