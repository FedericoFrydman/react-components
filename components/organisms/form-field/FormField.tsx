import { Children, forwardRef, ReactElement, Ref, useEffect, useState } from 'react';
import cn from 'classnames';

import './form-field.scss';
import { Text } from '../../atoms/text/Text';
import { Icon, IconNames } from '../../atoms/icon/Icon';
import { InputValidationStatus } from '../../molecules/input/Input';
import { IconColors } from '../../../components/assets/icons';

export interface FormFieldProps {
  /**
   * A unique identifier for the input field.
   */
  id?: string;
  /**
   * A clear and concise description of the form-field purpose.
   */
  label?: string;
  /**
   * A label to the left of the FormField
   */
  leftLabel?: string;
  /**
   * A label to the right of the FormField
   */
  rightLabel?: string;
  /**
   * Warning or error message.
   */
  validationHint?: string;
  /**
   * Triggers error | warning | success styles on the component. Important for accessibility.
   */
  status?: InputValidationStatus;
  /**
   * Brief description under the component
   */
  description?: string;
  /**
   * Removes the default margin below the text
   */
  noMargin?: boolean;
  /**
   * Children components
   */
  children?: ReactElement[] | ReactElement;
  /**
   * Stretch the component across the full width of its parent.
   */
  stretch?: boolean;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * Define if the input is required
   */
  required?: boolean;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLDivElement>;
}

/**
 * The FormField component is a container that adds the labels; validation messages, hints and colors for the different controls in a form (inputs, selects, checkbox, radio).
 */
const FormFieldComponent = (
  {
    id,
    label,
    leftLabel,
    rightLabel,
    validationHint,
    description,
    status,
    noMargin,
    children,
    stretch,
    classNames = [],
    required = false,
    ...props
  }: FormFieldProps,
  ref: FormFieldProps['ref'],
) => {
  const classFormField = 'lsux-form-field';
  const classContainer = `${classFormField}-container`;
  const cnContainer = cn(
    classContainer,
    {
      [`${classContainer}--stretch`]: stretch,
      [`${classContainer}--no-margin`]: noMargin,
    },
    classNames,
  );

  const cnValidation = {
    [`${classContainer}__hint_text`]: status == 'valid' || status == 'warning',
    [`${classContainer}__hint_text--invalid`]: status == 'invalid',
    ['pl-2']: true,
  };

  const [validationIcon, setValidationIcon] = useState<IconNames>(null);
  const [validationColor, setValidationColor] = useState<IconColors>(null);
  const [disabled, setDisabled] = useState(false);

  useEffect(() => {
    if (validationHint) {
      switch (status) {
        case 'valid':
          setValidationIcon('alert_success');
          setValidationColor('G200');
          break;
        case 'warning':
          setValidationIcon('alert_warning');
          setValidationColor('Y300');
          break;
        case 'invalid':
          setValidationIcon('alert_information');
          setValidationColor('R300');
          break;
        default:
          setValidationIcon(null);
          setValidationColor(null);
          break;
      }
    }
  }, [status, validationHint]);

  const elements = Children.toArray(children);

  useEffect(() => {
    if (elements.length > 0) {
      const element = elements[0] as ReactElement;
      setDisabled(element.props.disabled);
    }
  }, [elements]);

  return (
    <div {...props} ref={ref} className={cnContainer}>
      {label && (
        <label className={cn(`${classContainer}__label`, { disabled: disabled })} htmlFor={id}>
          {label}
          {required && <span className="required">*</span>}
        </label>
      )}
      {rightLabel || leftLabel ? (
        <div style={{ alignItems: 'center', display: 'flex' }}>
          {leftLabel && <div className="mr-3">{leftLabel}</div>}
          {children}
          {rightLabel && <div className="ml-3">{rightLabel}</div>}
        </div>
      ) : (
        <>{children}</>
      )}
      {description && (
        <Text textSize="small" classNames={[`${classContainer}__description`, 'pt-3']} text={description} />
      )}
      {validationHint && (
        <div className={cn(`${classContainer}__hint`, `pt-3`)}>
          <Icon name={validationIcon} color={validationColor} size="small" />
          <Text textSize="small" classNames={cnValidation} text={validationHint} />
        </div>
      )}
    </div>
  );
};

export const FormField = forwardRef(FormFieldComponent);
