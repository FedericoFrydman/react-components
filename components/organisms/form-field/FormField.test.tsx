import { render, screen } from '@testing-library/react';
import { shallow, ShallowWrapper } from 'enzyme';
import cn from 'classnames';

import { FormField, FormFieldProps } from './FormField';
import { Input } from '../../molecules/input/Input';
import { createRef } from 'react';

describe('FormField Component', () => {
  const testId = 'test-form-field';
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FormField id="input" label="text" data-testid={testId}>
        <Input />
      </FormField>,
    );
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('props', () => {
    describe('classNames', () => {
      describe('when provided classNames', () => {
        const classNames = ['my-class-1', 'my-class-2'];

        it('gives the component the provided classNames', () => {
          const formField = renderFormField({ classNames });

          expect(formField).toHaveClass(cn(classNames), { exact: false });
        });
      });
    });
  });

  describe('ref', () => {
    const ref = createRef<HTMLDivElement>();

    it('provides a ref to the FormField', () => {
      expect(renderFormField({ ref })).toEqual(ref.current);
    });
  });

  const renderFormField = (props: FormFieldProps = {}): HTMLElement => {
    render(
      <FormField {...props} data-testid={testId}>
        <Input />
      </FormField>,
    );

    return screen.getByTestId(testId);
  };
});
