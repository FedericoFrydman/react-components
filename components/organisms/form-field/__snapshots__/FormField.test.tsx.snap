// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField Component should render the component 1`] = `
<div
  className="lsux-form-field-container"
  data-testid="test-form-field"
>
  <label
    className="lsux-form-field-container__label"
    htmlFor="input"
  >
    text
  </label>
  <ForwardRef(InputComponent) />
</div>
`;
